### DISCUSSION TITLE
How to $set options on select after suffixAction

### DISCUSSION THREAD

**Discussion by AFSDARIF** [filamentphp/filament#7448] [Help]
![image](https://github.com/filamentphp/filament/assets/10229660/f6ea5915-97fb-4cf6-990e-25fd03349271)
i want create component select or update options select after suffixAction return.

**Comment by pboivin** [comment]
@AFSDARIF Please don't share screenshots of your code, share text instead. It's difficult to help by suggesting small modifications.

**Reply by AFSDARIF** [reply]
iam sorry. this is my code

private static $variant = [];

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('url')
                ->label('Url Product')
                ->columnSpan('full')
                ->suffixAction(fn ($component, $state, Closure $set) =>
                    Forms\Components\Actions\Action::make('search-action')
                    ->icon('heroicon-o-search')
                    ->size('lg')
                    ->action(function () use ($component, $state, $set) {
                        for($i=0;$i<5;$i++){
                            static::$variant[] = [
                                'variant' => 'cek',
                            ];
                        }

                        $component->getContainer()->getComponents()[1]->options(array_column(static::$variant,'variant'));
                        $set('variant', array_column(static::$variant,'variant'));
                        $set('name', 'full name');
                    }),
                ),

                Forms\Components\Select::make('variant')->reactive()->options(array_column(static::$variant,'variant')),
                Forms\Components\TextInput::make('name'),
            ]);
    }

**Reply by danharrin** [reply]
You cannot just "set" options, they are not state, they are configuration. The options should calculate themselves on-render, based on actual data that is being stored. Can you explain what you are trying to do?

**Reply by AFSDARIF** [reply]
![image](https://github.com/filamentphp/filament/assets/10229660/611c29bf-a9cf-466d-9e9f-fa4aef1dc0a8)
i want search data from url API and show it to component form, like select, textinput, image view and radio.

**Reply by danharrin** [reply]
The options() function should call the API and return the options
